api/src/main/java/com/epam/pipeline/manager/cluster/performancemonitoring/ResourceMonitoringManager.java

```diff
@@ -156,7 +160,26 @@ private void processOverloadedRuns(final List<PipelineRun> runs) {
                 .filter(pod -> isPodUnderPressure(pod.getValue(), thresholds))
                 .collect(Collectors.toList());
 
+        final List<PipelineRun> runsToUpdateTags = getRunsToUpdatePressuredTags(running, runsToNotify);
         notificationManager.notifyHighResourceConsumingRuns(runsToNotify, NotificationType.HIGH_CONSUMED_RESOURCES);
+        pipelineRunManager.updateRunsTags(runsToUpdateTags);
+    }
+
+    private List<PipelineRun> getRunsToUpdatePressuredTags(final Map<String, PipelineRun> running,
+                              final List<Pair<PipelineRun, Map<ELKUsageMetric, Double>>> runsToNotify) {
+        final Set<Long> listOfRunsIdToNotify = runsToNotify
+                .parallelStream()
```

Comment:
lets do it without parallel streams, since this code already starts in separate executor